gapis/api/vulkan/frame_loop.go

```diff
@@ -404,14 +416,82 @@ func (f *frameLoop) buildStartEndStates(ctx context.Context, startState *api.Glo
 				f.descriptorSetLayoutDestroyed[descriptorSetLayout] = true
 			}
 
-			// TODO: Recreate destroyed resources.
-		}
+		// Command Buffers
+		case *VkAllocateCommandBuffers:
+			vkCmd := cmd.(*VkAllocateCommandBuffers)
+			cmdBuffers := vkCmd.PCommandBuffers().MustRead(ctx, vkCmd, currentState, nil)
+			f.commandBufferAllocated[cmdBuffers] = true
+			log.D(ctx, "Command buffer %v allcoated.", cmdBuffers)
+		case *VkFreeCommandBuffers:
+			vkCmd := cmd.(*VkFreeCommandBuffers)
+			cmdBufCount := vkCmd.CommandBufferCount()
+			if cmdBufCount == 0 {
+				break
+			}
```

Comment:
Do we need these lines or will the below code just return and empty slice and iterate over nothing?

Reply:
yes you're right, just want to return earlier here, it can be can removed

Reply:
If the code has no functional effect, it just serves to confuse a reader and use screen space.